src/arduino.cc/builder/ctags/ctags_parser.go

```diff
@@ -229,7 +229,14 @@ func parseTag(row string) *types.CTag {
 	parts := strings.Split(row, "\t")
 
 	tag.FunctionName = parts[0]
-	tag.Filename = parts[1]
+	// This unescapes any backslashes in the filename. These
+	// filenames that ctags outputs originate from the line markers
+	// in the source, as generated by gcc. gcc escapes both
+	// backslashes and double quotes, but ctags ignores any escaping
+	// and just cuts off the filename at the first double quote it
+	// sees. This means any backslashes are still escaped, and need
+	// to be unescape, and any quotes will just break the build.
+	tag.Filename = strings.Replace(parts[1], "\\\\", "\\", -1)
```

Comment:
> If a double quote is present in a filename, it is escaped by gcc, but then ctags chops the string at the escaped double quote, causing the filename to end in a backslash. Parsing this by parseCppString will fail, so this would need extra special casing.

I see, probably we can just ignore `parseCppString` errors, and just use the filename as is (`parseCppString` already return the input filename unchanged if the parsing fails).
In any case a file containing a double quote will not work, but having the full parsing in place may allows future versions of ctags to run properly (without the need to touch this part again).